
Author: rmh
Status: sent to upstream
Description: Needed for "./config shared threads".  This is not done in Debian
  but it is in Gentoo.

diff -ur openssl-0.9.8a.old/Configure openssl-0.9.8a/Configure
--- openssl-0.9.8a.old/Configure	2005-08-02 10:59:42 +0000
+++ openssl-0.9.8a/Configure	2006-02-05 13:19:27 +0000
@@ -318,6 +318,7 @@
 #### IA-32 targets...
 "linux-ia32-icc",	"icc:-DL_ENDIAN -DTERMIO -O2 -no_cpprt::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-KPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-elf",	"gcc:-DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
+"glibc",	"gcc:-DL_ENDIAN -DTERMIOS -O3 -fomit-frame-pointer -Wall::-D_REENTRANT::-ldl:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_elf_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
 "linux-aout",	"gcc:-DL_ENDIAN -DTERMIO -O3 -fomit-frame-pointer -march=i486 -Wall::(unknown):::BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${x86_out_asm}",
 ####
 "linux-generic64","gcc:-DTERMIO -O3 -Wall::-D_REENTRANT::-ldl:SIXTY_FOUR_BIT_LONG RC4_CHAR RC4_CHUNK DES_INT DES_UNROLL BF_PTR:${no_asm}:dlfcn:linux-shared:-fPIC::.so.\$(SHLIB_MAJOR).\$(SHLIB_MINOR)",
diff -ur openssl-0.9.8a.old/config openssl-0.9.8a/config
--- openssl-0.9.8a.old/config	2005-09-20 12:16:30 +0000
+++ openssl-0.9.8a/config	2006-02-05 13:19:45 +0000
@@ -162,6 +162,10 @@
 	echo "${MACHINE}-whatever-linux1"; exit 0
 	;;
 
+    GNU/kFreeBSD:*)
+	echo "${MACHINE}-whatever-glibc"; exit 0;
+	;;
+
     GNU*)
 	echo "hurd-x86"; exit 0;
 	;;
@@ -601,6 +605,7 @@
         fi ;;
   *-*-linux1) OUT="linux-aout" ;;
   *-*-linux2) OUT="linux-generic32" ;;
+  *-*-glibc) OUT="glibc" ;;
   sun4u*-*-solaris2)
 	OUT="solaris-sparcv9-$CC"
 	ISA64=`(isalist) 2>/dev/null | grep sparcv9`
